runtime/litentry/src/migration/parachain_staking.rs

```diff
@@ -0,0 +1,748 @@
+			.map_err(|_| "Failed to decode Delayed Payouts")?;
+		for (round, actual_result) in <DelayedPayouts<T>>::iter() {
+			let expected_result: DelayedPayout<BalanceOf<T>> =
+				expected_state.get(&round).ok_or("Not Expected DelayedPayout")?.clone();
+			assert_eq!(expected_result.encode(), actual_result.encode());
+		}
+		Ok(())
+	}
+	pub fn pre_upgrade_staked_storage() -> Result<Vec<u8>, &'static str> {
+		let result: BTreeMap<u32, BalanceOf<T>> = <Staked<T>>::iter()
+			.map(|(round, state)| {
+				let new_staked: BalanceOf<T> = state;
+				(round, new_staked.saturating_mul(DECIMAL_CONVERTOR.into()))
+			})
+			.collect();
+		Ok(result.encode())
+	}
+	pub fn post_upgrade_staked_storage(state: Vec<u8>) -> Result<(), &'static str> {
+		let expected_state = BTreeMap::<u32, BalanceOf<T>>::decode(&mut &state[..])
+			.map_err(|_| "Failed to decode Staked")?;
+		for (round, actual_result) in <Staked<T>>::iter() {
+			let expected_result: BalanceOf<T> =
+				*expected_state.get(&round).ok_or("Not Expected DelayedPayout")?;
+			assert_eq!(expected_result.encode(), actual_result.encode());
+		}
+		Ok(())
+	}
+}
+
+impl<T> OnRuntimeUpgrade for ReplaceParachainStakingStorage<T>
+where
+	T: frame_system::Config + pallet_parachain_staking::Config,
+	BalanceOf<T>: From<u128>,
+{
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+		let delegator_state_vec = Self::pre_upgrade_delegator_state_storage()?;
+		let candidate_info_vec = Self::pre_upgrade_candidate_info_storage()?;
+		let delegation_scheduled_requests_vec =
+			Self::pre_upgrade_delegation_scheduled_requests_storage()?;
+		let top_delegations_vec = Self::pre_upgrade_top_delegations_storage()?;
+		let bottom_delegations_vec = Self::pre_upgrade_bottom_delegations_storage()?;
+		let total_vec = Self::pre_upgrade_total_storage()?;
+		let candidate_pool_vec = Self::pre_upgrade_candidate_pool_storage()?;
+		let delayed_payouts_vec = Self::pre_upgrade_delayed_payouts_storage()?;
+		let staked_vec = Self::pre_upgrade_staked_storage()?;
+		Ok((
+			delegator_state_vec,
+			candidate_info_vec,
+			delegation_scheduled_requests_vec,
+			top_delegations_vec,
+			bottom_delegations_vec,
+			total_vec,
+			candidate_pool_vec,
+			delayed_payouts_vec,
+			staked_vec,
+		)
+			.encode())
+	}
+
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		let mut weight = frame_support::weights::Weight::from_parts(0, 0);
+		weight += Self::replace_delegator_state_storage();
+		weight += Self::replace_candidate_info_storage();
+		weight += Self::replace_delegation_scheduled_requests_storage();
+		weight += Self::replace_top_delegations_storage();
+		weight += Self::replace_bottom_delegations_storage();
+		weight += Self::replace_total_storage();
+		weight += Self::replace_candidate_pool_storage();
+
+		// No need for AtStake Migration since this is a snapshot, everything is good as long as it
+		// will not change proportion AtStake
```

Comment:
It is used. But everything relying on it is calculated based on its proportion calculation. Like AtStaked storing two balance number A and B, and everywhere using A, B by (A divide B), which making the migration of AtStake old snapshot meaningless.

Reply:
You are correct, it does matter if collator is empty.

Reply:
I decide to skip this migration. 
On one side it will be only triggered when the collator empty and a round begin.
On the other side,
AtStake is a double map. Let's ignore it. It will introduce extra procedure of key concat.

Reply:
> On one side it will be only triggered when the collator empty and a round begin.

Will it cause problem once the set is _somehow_ empty?
Will this storage get updated / overwritten once after the runtime upgrade (e.g. during the block production phase)? We need to make sure it eventually reaches the right state